modules/dnn/src/layers/layers_common.cpp

```diff
@@ -55,128 +55,144 @@ std::string makeName(const std::string& str1, const std::string& str2)
-        util::getParameter(params, "pad", "pad", padT, padL, true, 0);
-        padB = padT;
-        padR = padL;
+        util::getParameter(params, "pad", "pad", pads_begin, true, 0);
+        if (pads_begin.size() == 2 || pads_begin.size() == 3)
+            pads_end = pads_begin;
+        else
+        {
+            pads_end = std::vector<size_t>(pads_begin.begin() + pads_begin.size() / 2, pads_begin.end());
+            pads_begin.resize(pads_begin.size() / 2);
+        }
     }
-    util::getParameter(params, "stride", "stride", strideH, strideW, true, 1);
+    util::getParameter(params, "stride", "stride", strides, true, 1);
 
     padMode = "";
     if (params.has("pad_mode"))
-    {
         padMode = params.get<String>("pad_mode");
-    }
 
-    CV_Assert(padT >= 0 && padL >= 0 && padB >= 0 && padR >= 0 && strideH > 0 && strideW > 0);
+    for (int i = 0; i < strides.size(); i++)
+        CV_Assert(strides[i] > 0);
 }
 }
 
-
-void getPoolingKernelParams(const LayerParams &params, int &kernelH, int &kernelW, bool &globalPooling,
-                            int &padT, int &padL, int &padB, int &padR, int &strideH, int &strideW, cv::String &padMode)
+void getPoolingKernelParams(const LayerParams &params, std::vector<size_t>& kernel, bool &globalPooling,
+                            std::vector<size_t>& pads_begin, std::vector<size_t>& pads_end,
+                            std::vector<size_t>& strides, cv::String &padMode)
 {
-    util::getStrideAndPadding(params, padT, padL, padB, padR, strideH, strideW, padMode);
-
+    util::getStrideAndPadding(params, pads_begin, pads_end, strides, padMode);
     globalPooling = params.has("global_pooling") &&
                     params.get<bool>("global_pooling");
-
     if (globalPooling)
     {
         if(params.has("kernel_h") || params.has("kernel_w") || params.has("kernel_size"))
         {
             CV_Error(cv::Error::StsBadArg, "In global_pooling mode, kernel_size (or kernel_h and kernel_w) cannot be specified");
         }
-        if(padT != 0 || padL != 0 || padB != 0 || padR != 0 || strideH != 1 || strideW != 1)
-        {
-            CV_Error(cv::Error::StsBadArg, "In global_pooling mode, pads must be = 0, and stride_h and stride_w must be = 1");
+        for (int i = 0; i < pads_begin.size(); i++) {
+            if (pads_begin[i] != 0 || pads_end[i] != 0)
+                CV_Error(cv::Error::StsBadArg, "In global_pooling mode, pads must be = 0");
+        }
+        for (int i = 0; i < strides.size(); i++) {
+            if (strides[i] != 1)
+                CV_Error(cv::Error::StsBadArg, "In global_pooling mode, strides must be = 1");
         }
     }
     else
     {
-        util::getKernelSize(params, kernelH, kernelW);
+        util::getKernelSize(params, kernel);
+        if (pads_begin.size() == kernel.size() - 1 && pads_end.size() == kernel.size() - 1) {
+            pads_begin.push_back(pads_begin.back());
+            pads_end.push_back(pads_end.back());
+        }
```

Comment:
I think that we need to remove 
```cpp
            if (parameter.size() == 1) {
                parameter.push_back(parameter.back());
            }
```
above and before the final checks do something like
```cpp
if (pads_begin.size() == 1) {
  pads_begin.resize(kernel.size(), pads_begin[0]);
}
```

Reply:
done